lib/PuppeteerSharp/BrowserFetcher.cs

```diff
@@ -261,7 +263,41 @@ private async Task<InstalledBrowser> DownloadAsync(SupportedBrowser browser, str
             await UnpackArchiveAsync(archivePath, outputPath, fileName).ConfigureAwait(false);
             new FileInfo(archivePath).Delete();
 
-            return new InstalledBrowser(cache, browser, buildId, Platform);
+            var installedBrowser = new InstalledBrowser(cache, browser, buildId, Platform);
+            RunSetup(installedBrowser);
+            return installedBrowser;
+        }
+
+        private void RunSetup(InstalledBrowser installedBrowser)
+        {
+            // On Windows for Chrome invoke setup.exe to configure sandboxes.
+            if (
+                installedBrowser.Platform is Platform.Win32 or Platform.Win64 &&
+                installedBrowser.Browser == SupportedBrowser.Chrome && installedBrowser.Platform == GetCurrentPlatform())
+            {
+                try
+                {
+                    var browserDir = new FileInfo(installedBrowser.GetExecutablePath()).Directory;
+                    var setupExePath = Path.Combine(browserDir!.FullName, "setup.exe");
+
+                    if (!File.Exists(setupExePath))
+                    {
+                        return;
+                    }
+
+                    using var process = new Process();
+                    process.StartInfo.FileName = setupExePath;
+                    process.StartInfo.Arguments = $"--configure-browser-in-directory=\"{browserDir.FullName}\"";
+                    process.StartInfo.RedirectStandardOutput = true;
+                    process.StartInfo.UseShellExecute = false;
+                    process.Start();
+                    process.WaitForExit();
+                }
+                finally
+                {
+                    // swallow
```

Comment:
What does a finally swallow?
Upstream just does some logging there: 
https://github.com/puppeteer/puppeteer/blob/79a3d049608056ab8dd50756e01103ba6e76b35c/packages/browsers/src/install.ts#L301

Reply:
Yeah. The main difference is that in puppeteer that's a CLI tool and here it's part of the code flow.

I added an optional logger :)